pkg/utils/validator/resourceoverride.go

```diff
@@ -102,9 +102,16 @@ func validateOverridePolicy(policy *fleetv1alpha1.OverridePolicy) error {
 				}
 			}
 		}
+		switch rule.OverrideType {
```

Comment:
can be replaced by using cel

for example,

// +kubebuilder:validation:XValidation:rule="self.overrideType == 'JSONPatch' || self.jsonPatchOverrides.size() == 0 ",message="jsonPatchOverrides must be empty when type is not JSONPatch"

Reply:
good to know but unless we remove this validator, we need to fix it even with the CEL, right?

Reply:
yeah, but the fix is much simpler, just need to add the type check.

Reply:
I think there is some advantage to keep the logic (and test) together intead of spliting them into two places.